src/main/scala/org/broadinstitute/dsde/firecloud/EntityClient.scala

```diff
@@ -84,21 +70,6 @@ class EntityClient (requestContext: RequestContext) extends Actor with FireCloud
   }
 
   /**
-   * Bail with a 400 Bad Request if the first column of the tsv has duplicate values.
-   * Otherwise, carry on. */
-  private def checkFirstColumnsDistinct( tsv: TSVLoadFile, tsvTypeName: String )(op: => Future[PerRequestMessage]): Future[PerRequestMessage] = {
-    val entitiesToUpdate = tsv.tsvData.map(_.headOption.get)
-    val distinctEntities = entitiesToUpdate.distinct
-    if ( entitiesToUpdate.size != distinctEntities.size ) {
-      Future( RequestCompleteWithErrorReport(BadRequest,
-        "Duplicated entities are not allowed in an " + tsvTypeName +
-          " TSV: " + entitiesToUpdate.diff(distinctEntities).distinct.mkString(", ")) )
-    } else {
-      op
-    }
-  }
-
-  /**
    * Bail with a 400 Bad Request if the tsv is trying to set members on a collection type.
    * Otherwise, carry on. */
   private def checkNoCollectionMemberAttribute( tsv: TSVLoadFile, memberTypeOpt: Option[String] )(op: => Future[PerRequestMessage]): Future[PerRequestMessage] = {
```

Comment:
My gut says yes - consolidating TSV operations in TSVFileSupport seems like the right move to me.